components/Buttons/src/MDCFloatingButton.m

```diff
@@ -71,55 +112,171 @@ - (instancetype)initWithFrame:(CGRect)frame shape:(MDCFloatingButtonShape)shape
+
+  BOOL isLeadingIcon = self.imageLocation == MDCFloatingButtonImageLocationLeading;
+  UIEdgeInsets adjustedLayoutInsets =
+      isLeadingIcon ? internalLayoutSpacingInsets
+                    : UIEdgeInsetsFlippedHorizonally(internalLayoutSpacingInsets);
+
+  const CGRect insetBounds = UIEdgeInsetsInsetRect(UIEdgeInsetsInsetRect(self.bounds,
+                                                                         adjustedLayoutInsets),
+                                                   self.contentEdgeInsets);
+
+  const CGFloat imageViewWidth = CGRectGetWidth(self.imageView.bounds);
+  const CGFloat boundsCenterY = CGRectGetMidY(insetBounds);
+  CGFloat titleWidthAvailable = CGRectGetWidth(insetBounds);
+  titleWidthAvailable -= imageViewWidth;
+  titleWidthAvailable -= self.imageTitleSpacing;
+
+  const CGFloat availableHeight = CGRectGetHeight(insetBounds);
+  CGSize titleIntrinsicSize
+      = [self.titleLabel sizeThatFits:CGSizeMake(titleWidthAvailable, availableHeight)];
+
+  const CGSize titleSize = CGSizeMake(MAX(0, MIN(titleIntrinsicSize.width, titleWidthAvailable)),
+                                      MAX(0, MIN(titleIntrinsicSize.height, availableHeight)));
+
+  CGPoint titleCenter;
+  CGPoint imageCenter;
+  BOOL isRTL = self.mdf_effectiveUserInterfaceLayoutDirection
+      == UIUserInterfaceLayoutDirectionRightToLeft;
+
+  // isRTL && isLeadingIcon => icon to right of text
+  // !isRTL && !isLeadingIcon => icon to right of text
+  // isRTL && !isLeadingIcon => icon to left of text
+  // !isRTL && isLeadingIcon => icon to left of text
+  if (isRTL == isLeadingIcon) {
+    const CGFloat imageCenterX
+        = CGRectGetMaxX(insetBounds) - (imageViewWidth / 2);
+    const CGFloat titleCenterX = CGRectGetMinX(insetBounds) + (titleWidthAvailable / 2);
+    imageCenter = CGPointMake(imageCenterX, boundsCenterY);
+    titleCenter = CGPointMake(titleCenterX, boundsCenterY);
+  } else {
+    const CGFloat imageCenterX
+        = CGRectGetMinX(insetBounds) + (imageViewWidth / 2);
+    const CGFloat titleCenterX
+        = CGRectGetMaxX(insetBounds) - (titleWidthAvailable / 2);
+    titleCenter = CGPointMake(titleCenterX, boundsCenterY);
+    imageCenter = CGPointMake(imageCenterX, boundsCenterY);
+  }
+
+  self.imageView.center = imageCenter;
+  self.imageView.frame = UIEdgeInsetsInsetRect(self.imageView.frame, self.imageEdgeInsets);
+  self.titleLabel.center = titleCenter;
+  CGRect newBounds = CGRectStandardize(self.titleLabel.bounds);
+  self.titleLabel.bounds = (CGRect){newBounds.origin, titleSize};
+  self.titleLabel.frame = UIEdgeInsetsInsetRect(self.titleLabel.frame, self.titleEdgeInsets);
 }
 
 #pragma mark - Subclassing
 
 - (UIEdgeInsets)defaultContentEdgeInsets {
-  switch (_shape) {
-    case MDCFloatingButtonShapeDefault:
-      return UIEdgeInsetsZero;
-    case MDCFloatingButtonShapeMini:
-      return UIEdgeInsetsMake(8, 8, 8, 8);
-  }
+  return UIEdgeInsetsZero;
```

Comment:
Will this change the behavior of any existing clients?

Reply:
I think it would if they're doing any blind arithmetic to supplement these. However, having all edge insets equal is equivalent to having them zero EXCEPT that it would start shrinking images, truncating text, etc.